app/components/Comments/CommentDisplay.jsx

```diff
@@ -36,6 +37,18 @@ export class CommentDisplay extends React.PureComponent {
     this.state = {isBlurred: false, showReplies: props.nesting !== 4}
   }
 
+  componentDidMount() {
```

Comment:
Two issues here:

1. We try to avoid calling `store.dispatch` directly. Prefer passing method during `connect` and access with `this.props.addStep`
2. This will have bad performances impact: if there are 60 comments on the page we're going to dispatch 60 actions on first mount

Reply:
1. Will do
2. I did this so that the event is registered as soon as it's component appears, for a modular approach. I do the unicity check in reducer. I was wondering about doing the check from a selector in the component but I think this breaks the MVC pattern as that puts business code in the view layer ? What do you think ?